sys/net/application_layer/dns/dns.c

```diff
@@ -95,24 +103,27 @@ static int _parse_dns_reply(uint8_t *buf, size_t len, void* addr_out, int family
 
     /* skip all queries that are part of the reply */
     for (unsigned n = 0; n < ntohs(hdr->qdcount); n++) {
-        bufpos += _skip_hostname(bufpos);
+        int tmp = _skip_hostname(buf, len, bufpos);
+        if (tmp < 0) {
+            return tmp;
+        }
+        bufpos += tmp;
         bufpos += 4;    /* skip type and class of query */
     }
 
     for (unsigned n = 0; n < ntohs(hdr->ancount); n++) {
-        bufpos += _skip_hostname(bufpos);
+        int tmp = _skip_hostname(buf, len, bufpos);
+        if (tmp < 0) {
+            return tmp;
+        }
+        bufpos += tmp;
         uint16_t _type = ntohs(_get_short(bufpos));
```

Comment:
As you said: unrelated to this PR, but I will make a mental note to enhance in a later PR.

Reply:
There's actually ```byteorder_bebuftohs()``` in byteorder.h doing exactly that.

Reply:
Always wondered, why you didn't use `byteorder` here anyways ;-)

Reply:
> why you didn't use `byteorder` here anyways ;-)

This code was initially written for sock on linux, then ported to RIOT...

Reply:
Is it still a required to run on Linux?

Reply:
It would be nice, but the linux version can also simply import RIOT's byteorder file. If not, don't worry about it now, I'll work around somehow if necessary.